### PR TITLE
feat/app-deployment-plugin 

### PR DESCRIPTION
**Pull request by athul-rs**
## What

- App Deployment moved out of the code base.

## Why

- Needs to be a plugin feature.

## How

- Converting earlier used code to be present as a plugin and using a feature flag
- Add generic APIs to list and evaluate feature flags

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- Yes, codes are removed from frontend.

## Database Migrations

- N/A

## Env Config

- N/A

## Relevant Docs

- N/A

## Related Issues or PRs

- N/A

## Dependencies Versions

- N/A

## Notes on Testing

- Done an app run-up and tested basic functionality after removal of the app deployment code.

## Screenshots

image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by vishnuszipstack — APPROVED** [review]
LGTM

File: frontend/src/components/navigations/side-nav-bar/SideNavBar.jsx [hunk 1]
```diff
@@ -147,6 +144,10 @@ const SideNavBar = ({ collapsed }) => {
     },
   ];
 
+  if (getMenuItem && appDeployment) {
+    data[0].subMenu.splice(1, 0, getMenuItem.default(orgName));
```
Comment (johnyrahul) [inline frontend/src/components/navigations/side-nav-bar/SideNavBar.jsx:148]: why do we require orgaName as a parameter here.? @athul-rs

**Review by johnyrahul — APPROVED** [review]
LGTM

File: frontend/src/components/navigations/side-nav-bar/SideNavBar.jsx (hunk 1 above)
Comment (athul-rs) [reply frontend/src/components/navigations/side-nav-bar/SideNavBar.jsx:148]: Using this in the plugin  to call the backend url where `org_id` is required.

File: frontend/src/hooks/useSessionValid.js [hunk 2]
```diff
@@ -105,6 +106,16 @@ function useSessionValid() {
         if (remainingTrialDays)
           userAndOrgDetails["remainingTrialDays"] = remainingTrialDays;
       }
+
+      if (getFlagStatus) {
+        const appDeploymentFlag = await getFlagStatus.fetchFeatureFlagStatus(
+          orgId,
+          csrfToken,
+          "app_deployment"
```
Comment (athul-rs) [inline frontend/src/hooks/useSessionValid.js:114]: @hari-kuriakose Quick doubt, should we keep the flag name as such or convert it to a env in front-end ?

File: unstract/flags/src/unstract/flags/clients/base_client.py
Comment (hari-kuriakose) [inline unstract/flags/src/unstract/flags/clients/base_client.py:1]: @athul-rs The folder and file name can be `flags/client/base.py` instead.

File: unstract/flags/src/unstract/flags/clients/evaluation_client.py
Comment (hari-kuriakose) [inline unstract/flags/src/unstract/flags/clients/evaluation_client.py:1]: @athul-rs The folder and file name can be `flags/client/evaluation.py` instead.

File: unstract/flags/src/unstract/flags/clients/flipt_client.py
Comment (hari-kuriakose) [inline unstract/flags/src/unstract/flags/clients/flipt_client.py:1]: @athul-rs The folder and file name can be `flags/client/flipt.py` instead.

File: unstract/flags/src/unstract/flags/protos/flipt.proto
Comment (hari-kuriakose) [inline unstract/flags/src/unstract/flags/protos/flipt.proto:1]: @athul-rs The folder and file name can be `flags/proto/flipt.proto` instead.

File: unstract/flags/src/unstract/flags/clients/flipt_client.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import logging
+
+import grpc
+
+from ..generated import flipt_pb2, flipt_pb2_grpc
+from .base_client import BaseClient
+
+
+class FliptClient(BaseClient):
```
Comment (hari-kuriakose) [inline unstract/flags/src/unstract/flags/clients/flipt_client.py:9]: @athul-rs Good to add a doc string mentioning the purpose and/or how this client is different from others.

File: unstract/flags/src/unstract/flags/clients/evaluation_client.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+from typing import Optional
+
+import grpc
+
+from ..generated import evaluation_pb2, evaluation_pb2_grpc
+from .base_client import BaseClient
+
+
+class EvaluationClient(BaseClient):
```
Comment (hari-kuriakose) [inline unstract/flags/src/unstract/flags/clients/evaluation_client.py:9]: @athul-rs Good to add a doc string mentioning the purpose and/or how this client is different from others.

**Review by hari-kuriakose — APPROVED** [review]
@athul-rs LGTM overall.

Added few minor comments though.

File: unstract/flags/src/unstract/flags/clients/base_client.py
Comment (athul-rs) [reply unstract/flags/src/unstract/flags/clients/base_client.py:1]: updated

File: unstract/flags/src/unstract/flags/clients/evaluation_client.py
Comment (athul-rs) [reply unstract/flags/src/unstract/flags/clients/evaluation_client.py:1]: updated

File: unstract/flags/src/unstract/flags/clients/flipt_client.py
Comment (athul-rs) [reply unstract/flags/src/unstract/flags/clients/flipt_client.py:1]: updated

File: unstract/flags/src/unstract/flags/protos/flipt.proto
Comment (athul-rs) [reply unstract/flags/src/unstract/flags/protos/flipt.proto:1]: updated

File: unstract/flags/src/unstract/flags/clients/flipt_client.py (hunk 3 above)
Comment (athul-rs) [reply unstract/flags/src/unstract/flags/clients/flipt_client.py]: updated

Comment (athul-rs) [reply unstract/flags/src/unstract/flags/clients/flipt_client.py]: updated

**Pull request merged**